pkg/spire/spire_test.go

```diff
@@ -356,86 +368,74 @@ func TestSpire_TaskRunResultsSign(t *testing.T) {
-		// List of PipelineResourceResult to test against
-		pipelineResourceResults [][]v1beta1.PipelineResourceResult
-	}{
-		{
-			desc:            "sign/verify result when entry isn't created",
-			skipEntryCreate: true,
-			success:         false,
-			// Using single taskrun and pipelineResourceResults as the unit test
-			// times out with the default 30 second. getWorkloadSVID has a 20 second
-			// time out before it throws an error for no svid found
-			taskRunList:             testSingleTaskRun(),
-			pipelineResourceResults: testSinglePipelineResourceResults(),
-		},
-		{
-			desc:                    "regular sign/verify result",
-			success:                 true,
-			taskRunList:             testTaskRuns(),
-			pipelineResourceResults: testPipelineResourceResults(),
-		},
-		{
-			desc:                    "sign/verify result when signing with wrong pod identity",
-			wrongPodIdentity:        true,
-			success:                 false,
-			taskRunList:             testTaskRuns(),
-			pipelineResourceResults: testPipelineResourceResults(),
-		},
-	}
+	trList := testTaskRuns()
+	rsrcResults := testPipelineResourceResults()
 
-	for _, tt := range testCases {
-		ctx := context.Background()
-		for _, tr := range tt.taskRunList {
-			if !tt.skipEntryCreate {
-				if !tt.wrongPodIdentity {
-					resp := &fakeworkloadapi.X509SVIDResponse{
-						Bundle: ca.X509Bundle(),
-						SVIDs:  makeX509SVIDs(ca, spiffeid.RequireFromPath(td, getTaskrunPath(tr))),
-					}
-					wl.SetX509SVIDResponse(resp)
-				} else {
-					resp := &fakeworkloadapi.X509SVIDResponse{
-						Bundle: ca.X509Bundle(),
-						SVIDs:  makeX509SVIDs(ca, fooID),
-					}
-					wl.SetX509SVIDResponse(resp)
+	wl.SetX509SVIDResponse(&fakeworkloadapi.X509SVIDResponse{
+		Bundle: ca.X509Bundle(),
+		SVIDs:  x509svids(ca, spiffeid.RequireFromPath(td, "/bogus")),
+	})
+
+	for _, tr := range trList {
+		for _, results := range rsrcResults {
+			if sigResults, err := ec.Sign(ctx, results); err != nil {
+				t.Errorf("ec.Sign error: %v", err)
+			} else {
+				results = append(results, sigResults...)
+				// It would be nice to check the error type, but
+				// VerifyTaskRunResults returns a generic errors.New().
+				if cc.VerifyTaskRunResults(ctx, results, tr) == nil {
+					t.Error("got nil; want error")
 				}
 			}
+		}
+	}
+}
 
-			for _, results := range tt.pipelineResourceResults {
-				success := func() bool {
-					sigResults, err := ec.Sign(ctx, results)
-					if err != nil {
-						return false
-					}
+func TestSignTaskRunResults(t *testing.T) {
+	ctx, _ := ttesting.SetupDefaultContext(t)
 
-					results = append(results, sigResults...)
+	wl := fakeworkloadapi.New(t)
+	defer wl.Stop()
+
+	ca := test.NewCA(t, td)
+	wl.SetX509Bundles(ca.X509Bundle())
+
+	cfg := &config.SpireConfig{
+		SocketPath:  wl.Addr(),
+		TrustDomain: trustDomain,
+	}
+
+	ec := NewEntrypointerAPIClient(cfg)
+	cc := GetControllerAPIClient(ctx)
+	cc.SetConfig(*cfg)
+	defer cc.Close()
+	defer ec.Close()
```

Comment:
*extremely* optional: Almost every test in this file has this same setup. Is it worth factoring this out into a test helper?

Reply:
not today 😸
/meow